src/components/cuda_pw/common.h

```diff
@@ -0,0 +1,36 @@
+/*
+  This file contains data structures visible to linux-cuda.c and cupti_profiler.h
+*/
+
+#ifndef __COMMON_H__
+#define __COMMON_H__
+
+#include <papi.h>
+
+typedef struct eventname_id_s {
+    char name[PAPI_2MAX_STR_LEN];
+    char desc[PAPI_2MAX_STR_LEN];
+    int gpu_id;
+    unsigned int evt_code;
+    unsigned int evt_pos;  // index of added event
+    int num_dep;
+    double value;
+    void * info;  // API specific details
+} event_rec_t;
```

Comment:
According to object oriented programming, objects can have public members and private members. I intended `void *info` to be the private part, while the public stuff are accessible where needed, front or back end. For instance, `value` member stores measured value per event per device per thread in the low level, and `evt_pos` tells the top level which event index in the control state that value corresponds to. It makes the design quite transparent in my opinion as to what is going on.

Also, `info` here is only intended for the events api because it needs types like `CUpti_eventID` etc, the profiler api does not use it.